sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1428,124 +1429,21 @@ class Analyzer(override val catalogManager: CatalogManager)
-          Seq((oldVersion, oldVersion.copy(output = output.map(_.newInstance()))))
-
-        case oldVersion @ FlatMapCoGroupsInPandas(_, _, _, output, _, _)
-            if oldVersion.outputSet.intersect(conflictingAttributes).nonEmpty =>
-          Seq((oldVersion, oldVersion.copy(output = output.map(_.newInstance()))))
-
-        case oldVersion @ MapInPandas(_, output, _)
-            if oldVersion.outputSet.intersect(conflictingAttributes).nonEmpty =>
-          Seq((oldVersion, oldVersion.copy(output = output.map(_.newInstance()))))
-
-        case oldVersion: Generate
-            if oldVersion.producedAttributes.intersect(conflictingAttributes).nonEmpty =>
-          val newOutput = oldVersion.generatorOutput.map(_.newInstance())
-          Seq((oldVersion, oldVersion.copy(generatorOutput = newOutput)))
-
-        case oldVersion: Expand
-            if oldVersion.producedAttributes.intersect(conflictingAttributes).nonEmpty =>
-          val producedAttributes = oldVersion.producedAttributes
-          val newOutput = oldVersion.output.map { attr =>
-            if (producedAttributes.contains(attr)) {
-              attr.newInstance()
-            } else {
-              attr
-            }
-          }
-          Seq((oldVersion, oldVersion.copy(output = newOutput)))
-
-        case oldVersion @ Window(windowExpressions, _, _, child)
-            if AttributeSet(windowExpressions.map(_.toAttribute)).intersect(conflictingAttributes)
-            .nonEmpty =>
-          Seq((oldVersion, oldVersion.copy(windowExpressions = newAliases(windowExpressions))))
 
-        case oldVersion @ ScriptTransformation(_, _, output, _, _)
-            if AttributeSet(output).intersect(conflictingAttributes).nonEmpty =>
-          Seq((oldVersion, oldVersion.copy(output = output.map(_.newInstance()))))
-
-        case _ => plan.children.flatMap(collectConflictPlans)
-      }
-
-      val conflictPlans = collectConflictPlans(right)
-
-      /*
-       * Note that it's possible `conflictPlans` can be empty which implies that there
-       * is a logical plan node that produces new references that this rule cannot handle.
-       * When that is the case, there must be another rule that resolves these conflicts.
-       * Otherwise, the analysis will fail.
-       */
-      if (conflictPlans.isEmpty) {
-        right
-      } else {
-        val planMapping = conflictPlans.toMap
-        right.transformUpWithNewOutput {
-          case oldPlan =>
-            val newPlanOpt = planMapping.get(oldPlan)
-            newPlanOpt.map { newPlan =>
-              newPlan -> oldPlan.output.zip(newPlan.output)
-            }.getOrElse(oldPlan -> Nil)
-        }
+    private def hasConflictingAttrs(p: LogicalPlan): Boolean = {
+      p.children.length > 1 && {
+        p.children.tail.foldLeft(p.children.head.outputSet) {
+          case (conflictAttrs, child) => conflictAttrs.intersect(child.outputSet)
```

Comment:
when can one operator have duplicated attr IDs in its output?

Reply:
For example, the test `SQLQuerySuite.self join with alias in agg`:

```scala
[info]   org.apache.spark.sql.AnalysisException: cannot resolve 'x.str' given input columns: [x.str, y.str, x.strCount, y.strCount]; line 3 pos 23;
[info]   'Aggregate ['x.str], ['x.str, unresolvedalias('SUM('x.strCount), None)]
[info]   +- 'Join Inner, ('x.str = 'y.str)
[info]      :- SubqueryAlias x
[info]      :  +- SubqueryAlias df
[info]      :     +- View (`df`, [str#226,str#226,strCount#233L])
[info]      :        +- Aggregate [str#226], [str#226, str#226, count(str#226) AS strCount#233L]
[info]      :           +- Project [_1#220 AS int#225, _2#221 AS str#226]
[info]      :              +- LocalRelation [_1#220, _2#221]
[info]      +- SubqueryAlias y
[info]         +- SubqueryAlias df
[info]            +- View (`df`, [str#241,str#241,strCount#239L])
[info]               +- Aggregate [str#241], [str#241, str#241, count(str#241) AS strCount#239L]
[info]                  +- Project [_1#237 AS int#240, _2#238 AS str#241]
[info]                     +- LocalRelation [_1#237, _2#238]
```

Here, the `Aggregate` has duplicate `str`s.